### DISCUSSION TITLE
istoreos占用的443端口，是干啥的？我想让nginx占用443

### DISCUSSION THREAD

**Discussion by SyberRabbit** [istoreos/istoreos#1996] [Q&A] [answered]
这是我找到的占用的443的程序信息，这是啥？
可以关掉吗？或者，这有nginx类似的功能吗？
> /usr/sbin/uhttpd -f -h /www -r iStoreOS -x /cgi-bin -u /ubus -t 3600 -T 30 -k 20 -A 1 -n 50 -N 100 -R -p 0.0.0.0:80 -p [::]:80 -C /etc/uhttpd.crt -K /etc/uhttpd.key -s 0.0.0.0:443 -s [::]:443

![图片](https://github.com/user-attachments/assets/5d3f005f-2a6c-4755-a9ab-fcaae4b57b23)

**Comment by jjm2473** [comment, marked as answer]
就是luci后台页面啊，由uhttpd提供服务。可以修改uhttpd的端口，在istore安装uhttpd就有管理界面了，可以修改端口。之后你的nginx就可以使用443，80端口了。

**Reply by docker-master** [reply]
在uhttpd把443改成4433剩余一个80端口给IPV6的动态DNS用
<img width="1794" height="1516" alt="image" src="https://github.com/user-attachments/assets/5cc3c344-02f2-4b26-aed9-a085acd78585" />
lucky大吉申请ssl证书很方便，使用443正好，完美
<img width="1759" height="1104" alt="image" src="https://github.com/user-attachments/assets/d9674de7-c812-4510-a383-c11ecbe472ee" />

**Comment by w9315273** [comment]
会用Nginx的话建议直接把uhttpd删了 全面换成Nginx
```
/etc/init.d/uhttpd stop && \
/etc/init.d/uhttpd disable && \
opkg remove --force-removal-of-dependent-packages uhttpd && \
rm /etc/config/uhttpd /etc/uhttpd.crt /etc/uhttpd.key && \
opkg update && \
opkg install nginx-all-module \
nginx-mod-luci \
uwsgi \
uwsgi-cgi-plugin \
uwsgi-luci-support \
luci-ssl-nginx \
libopenssl-conf \
openssl-util \
uwsgi-syslog-plugin
```

**Reply by SyberRabbit** [reply]
感谢提供方法，我是会一些nginx，但linux这块，nginx组件和istoreos拼装集成这一块，就有点知识盲区了，我想换nginx纯粹是强迫症，，uhttp我就凑合用吧，，

**Reply by w9315273** [reply]
uhttpd基本什么功能都没有 这个打包命令运行就替换完了 uci配置在`/etc/config/nginx` nginx目录在`/etc/nginx` 这是知道的吧

**Reply by jjm2473** [reply]
换uhttpd的端口就行了，万一nginx挂掉还能访问luci后台

**Comment by SyberRabbit** [comment]
补充记录一下，我将uhttps的80和443端口，改了。
然后在istoreos的商店一键安装nginx，中间出现日志错误，原网站也打不开了，之后等会就好了，可以用80来访问istoreos了，但是我并没有在ui界面中找到nginx的设置，难不成，商店安装的nginx，也是手动修改conf来使用的吗？
实在找不到设置，也没有相关教程，我就又卸载了。
转头去docker上安装了Nginx Proxy Manager
目前正常

**Reply by w9315273** [reply]
Nginx当然要自己配置了
你说的那个软件不好用 只提供很基础的功能 不过你可以当作一个入门

**Reply by jjm2473** [reply]
nginx没有界面配置

**Reply by SyberRabbit** [reply]
> nginx没有界面配置

我知道nginx没有，我也会配置conf，但是和istorceos集成后，在os上无法控制nginx的话，这和我单独安装一个nginx有啥区别。。
而我在istoreos的路由器上使用nginx业务也不复杂，ui界面的够用了
